delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/auth/KeycloakAuthService.scala

```diff
@@ -0,0 +1,70 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.auth
+
+import akka.http.javadsl.model.headers.HttpCredentials
+import akka.http.scaladsl.model.HttpMethods.POST
+import akka.http.scaladsl.model.headers.Authorization
+import akka.http.scaladsl.model.{HttpRequest, Uri}
+import cats.effect.Clock
+import ch.epfl.bluebrain.nexus.delta.kernel.Secret
+import ch.epfl.bluebrain.nexus.delta.kernel.effect.migration.MigrateEffectSyntax
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.IOUtils
+import ch.epfl.bluebrain.nexus.delta.sdk.auth.AuthMethod.Credentials
+import ch.epfl.bluebrain.nexus.delta.sdk.error.TokenError.{ExpiryNotFoundInResponse, TokenHttpError, TokenNotFoundInResponse}
+import ch.epfl.bluebrain.nexus.delta.sdk.http.HttpClient
+import ch.epfl.bluebrain.nexus.delta.sdk.realms.Realms
+import io.circe.Json
+import monix.bio.{IO, UIO}
+
+import java.time.{Duration, Instant}
+
+class KeycloakAuthService(httpClient: HttpClient, realms: Realms)(implicit clock: Clock[UIO])
```

Comment:
Drop the keycloak term to be consistent with the other parts of the codebase and the documentation.
We also realms depending on github for example